src/cargo/core/compiler/build_runner/mod.rs

```diff
@@ -446,6 +456,44 @@ impl<'a, 'gctx> BuildRunner<'a, 'gctx> {
         self.files().metadata(unit).unit_id()
     }
 
+    /// Returns the list of SBOM output file paths for a given [`Unit`].
+    ///
+    /// Only call this function when `sbom` is active.
+    pub fn sbom_output_files(&self, unit: &Unit) -> CargoResult<Vec<PathBuf>> {
+        const SBOM_FILE_EXTENSION: &str = ".cargo-sbom.json";
+
+        fn append_sbom_suffix(link: &PathBuf, suffix: &str) -> PathBuf {
+            let mut link_buf = link.clone().into_os_string();
+            link_buf.push(suffix);
+            PathBuf::from(link_buf)
+        }
+
+        assert!(self.bcx.build_config.sbom);
+        let sbom_enabled = match unit.target.kind() {
+            TargetKind::Lib(crate_types) | TargetKind::ExampleLib(crate_types) => {
+                crate_types.iter().any(|crate_type| {
+                    matches!(
+                        crate_type,
+                        CrateType::Cdylib | CrateType::Dylib | CrateType::Staticlib
+                    )
+                })
+            }
+            TargetKind::Bin | TargetKind::Test | TargetKind::Bench | TargetKind::ExampleBin => true,
+            TargetKind::CustomBuild => false,
+        };
+        if !sbom_enabled {
+            return Ok(Vec::new());
+        }
+
+        let files = self
+            .outputs(unit)?
+            .iter()
+            .filter(|o| matches!(o.flavor, FileFlavor::Normal | FileFlavor::Linkable))
```

Comment:
Do we have a place to track specific points we want to have called out in the RFC?  For me, when we do this is one of them

Reply:
I've added a list to the PR description